halo/lua/lua.cpp

```diff
@@ -4,36 +4,88 @@
  * This program is free software under the GNU General Public License v3.0 or later. See LICENSE for more information.
  */
 
-#include "lua.hpp"
-#include "../memory/types.hpp"
-#include "../memory/global.hpp"
+#include <string>
 
-#define WIN32_MEAN_AND_LEAN
-#include <windows.h>
-#include <cstdio>
+#include <lua.hpp>
 
+#include "../../includes/file_helpers.hpp"
+#include "../hooks/map.hpp"
+#include "../memory/global.hpp"
+#include "../memory/types.hpp"
 #include "../paths.hpp"
 
+#include "lua_helpers.hpp"
+#include "lua_messaging.hpp"
+#include "lua_open.hpp"
+#include "lua.hpp"
+
+static lua_State* map_state = NULL;
+
 static bool initialized = false;
 
+const auto INDEX = std::string("index.lua");
+
+static void luaV_register_functions(lua_State *state, bool sandboxed){
+    luaV_openlibs(state, !sandboxed);
+    luaV_reg_messaging_funcs(state);
+}
+
+//TODO: Make this load scripts from maps.
+static void luaV_load_scripts_for_map(){
+    // Get path to where the lua scripts should be stored.
+    auto cur_map_name = std::string(map_name());
+    auto path_str = std::string(profile_path()) + LUA_MAP_PATH + "\\" + cur_map_name + "\\";
+
+    // Check if it is actually a directory / even exists.
+    if (is_dir(path_str)){
+        // Check if there is an index.lua
+        auto file_str = path_str + INDEX;
+
+        if (is_file(file_str)){
```

Comment:
This is more to short circuit the creation of the lua state. If it gets deleted before the next part the code is executed it will still cancel the creation gracefully, just a little later.